src/analyses/goto_check_c.cpp

```diff
@@ -340,6 +340,82 @@ class goto_check_ct
   named_check_statust match_named_check(const irep_idt &named_check) const;
 };
 
+/// Allows to:
+/// - override a Boolean flag with a new value via `set_flag`
+/// - set a Boolean flag to false via `disable_flag`, such that
+///   previous `set_flag` are overridden and future `set_flag` are ignored.
+///
+/// A flag's initial value (before any `set_flag` or `disable_flag`) is restored
+/// when the entire object goes out of scope.
+class flag_resett
```

Comment:
I realise this is just code you've moved, but just wondering about the naming of this class (sorry...) Conceptually what this class is really about is less about "resetting" a flag, and more about introducing a "scope" for the flag? So maybe call it something like `flag_scopet` ? or `flag_overridet` ? One for a later PR though.

Reply:
Good call, I think `flag_overridet` is the best match for what it does. I've inserted a commit to do the renaming.